monstache.go

```diff
@@ -1107,6 +1129,15 @@ func (ic *indexClient) processRelated(root *gtm.Op) (err error) {
 					continue
 				}
 
+				if r.MatchFieldType == "objectId" {
```

Comment:
Can you extract this part into one more function, e.g. `convertSrcDataToMatchFieldType` like

```golang
if r.MatchFieldType != "" {
if srcData, err = convertSrcDataToMatchFieldType(srcData, r.MatchFieldtype); err != nil {
						ic.processErr(err)
						continue
					}
}
```
This new function `convertSrcDataToMatchFieldType` can then check which of `convertSrcDataToObjectId` or `convertSrcDataToString` to invoke.